### PR TITLE
[v6r22] Adding OwnerDN and OwnerGroup for DelayedExternalDeletion

### PR DESCRIPTION
**Pull request by fstagni**
For #4212 

BEGINRELEASENOTES

*WMS
CHANGE: Sandbox: Adding OwnerDN and OwnerGroup for DelayedExternalDeletion

ENDRELEASENOTES

### CODE REVIEW

File: DataManagementSystem/Client/FailoverTransfer.py [hunk 1]
```diff
@@ -232,32 +243,41 @@ def _setRegistrationRequest( self, lfn, targetSE, fileDict, catalog ):
...
-      regFile.GUID = fileDict.get( "GUID", "" )
+      regFile.Checksum = fileDict.get("Checksum", "")
+      regFile.ChecksumType = fileDict.get("ChecksumType", self.defaultChecksumType)
+      regFile.Size = fileDict.get("Size", 0)
+      regFile.GUID = fileDict.get("GUID", "")
 
-      se = StorageElement( targetSE )
-      pfn = se.getURL( lfn, self.registrationProtocols )
+      se = StorageElement(targetSE)
+      pfn = se.getURL(lfn, self.registrationProtocols)
```
Comment (andresailer) [inline DataManagementSystem/Client/FailoverTransfer.py]: use the `returnSingleResult` helper and simplify the code below

File: WorkloadManagementSystem/Service/SandboxStoreHandler.py [hunk 2]
```diff
@@ -461,13 +461,13 @@ def __deleteSandboxFromBackend(self, SEName, SEPFN):
         gLogger.error("Cannot delete local sandbox", "%s : %s" % (hdPath, repr(e).replace(',)', ')')))
       while hdPath:
         hdPath = os.path.dirname(hdPath)
-        gLogger.info("Checking if dir %s is empty" % hdPath)
+        gLogger.info("Checking if dir is empty", "%s" % hdPath)
```
Comment (andresailer) [inline WorkloadManagementSystem/Service/SandboxStoreHandler.py]:
```suggestion
        gLogger.info("Checking if dir is empty", hdPath)
```

File: WorkloadManagementSystem/Service/SandboxStoreHandler.py [hunk 3]
```diff
@@ -461,13 +461,13 @@ def __deleteSandboxFromBackend(self, SEName, SEPFN):
...
         try:
           if not os.path.isdir(hdPath):
             break
           if os.listdir(hdPath):
             break
-          gLogger.info("Trying to clean dir %s" % hdPath)
+          gLogger.info("Trying to clean dir", "%s" % hdPath)
```
Comment (andresailer) [inline WorkloadManagementSystem/Service/SandboxStoreHandler.py]:
```suggestion
          gLogger.info("Trying to clean dir", hdPath)
```

File: DataManagementSystem/Client/FailoverTransfer.py [hunk 4]
```diff
@@ -232,32 +248,32 @@ def _setRegistrationRequest( self, lfn, targetSE, fileDict, catalog ):
...
-      if not pfn["OK"] or lfn not in pfn["Value"]['Successful']:
-        self.log.error( "Unable to get PFN for LFN", "%s" % pfn.get( 'Message', pfn.get( 'Value', {} ).get( 'Failed', {} ).get( lfn ) ) )
-        return pfn
-      regFile.PFN = pfn["Value"]['Successful'][lfn]
+      se = StorageElement(targetSE)
+      res = returnSingleResult(se.getURL(lfn, self.registrationProtocols))
+      if not res["OK"]:
+        self.log.error("Unable to get PFN for LFN", res['Message'])
+        return res
+      regFile.PFN = res["Value"][lfn]
```
Comment (andresailer) [inline DataManagementSystem/Client/FailoverTransfer.py]: I think this has to be
```suggestion
      regFile.PFN = res["Value"]
```

Comment (fstagni) [reply DataManagementSystem/Client/FailoverTransfer.py]: Right, corrected.

**Pull request merged**